lib/pacemaker/pcmk_fence.c

```diff
@@ -240,12 +240,23 @@ pcmk__request_fencing(stonith_t *st, const char *target, const char *action,
 }
 
 int
-pcmk_request_fencing(stonith_t *st, const char *target, const char *action,
-                     const char *name, unsigned int timeout,
-                     unsigned int tolerance, int delay, char **reason)
+pcmk_request_fencing(const char *target, const char *action, const char *name,
```

Comment:
I think the high-level public APIs should always take an XML output argument for consistency and future-proofing, even if it just has the exit status.

Reply:
Yeah, I went back and forth on that too but I think you're right.  This also eliminates the need for the setup_fencing function, and less code is always better.